receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -204,6 +327,86 @@ func (a alertsReceiver) Shutdown(ctx context.Context) error {
 	return nil
 }
 
+func (a alertsReceiver) shutdownPoller(ctx context.Context) error {
+	a.logger.Debug("Shutting down client")
+	close(a.doneChan)
+	a.wg.Wait()
+	return a.writeCheckpoint(ctx)
+}
+
+func (a alertsReceiver) convertAlerts(now pcommon.Timestamp, alerts []mongodbatlas.Alert) (plog.Logs, error) {
+	logs := plog.NewLogs()
+	var errs error
+	for _, alert := range alerts {
+		resourceLogs := logs.ResourceLogs().AppendEmpty()
+		logRecord := resourceLogs.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+		logRecord.SetObservedTimestamp(now)
+
+		ts, err := time.Parse(time.RFC3339, alert.Updated)
+		if err != nil {
+			a.logger.Warn("unable to interpret updated time for alert, expecting a RFC3339 timestamp", zap.String("timestamp", alert.Updated))
+			continue
+		}
+
+		logRecord.SetTimestamp(pcommon.NewTimestampFromTime(ts))
+		logRecord.SetSeverityNumber(severityFromAPIAlert(alert))
+		// this could be fairly expensive to do, expecting not too many issues unless there are a ton
+		// of unrecognized alerts to process.
+		bodyBytes, err := json.Marshal(alert)
+		if err != nil {
+			a.logger.Warn("unable to marshal alert into a body string")
+			continue
+		}
+
+		logRecord.Body().SetStr(string(bodyBytes))
+
+		resourceAttrs := resourceLogs.Resource().Attributes()
+		resourceAttrs.PutString("mongodbatlas.group.id", alert.GroupID)
+		resourceAttrs.PutString("mongodbatlas.alert.config.id", alert.AlertConfigID)
+		putStringToMapNotNil(resourceAttrs, "mongodbatlas.cluster.name", &alert.ClusterName)
+		putStringToMapNotNil(resourceAttrs, "mongodbatlas.replica_set.name", &alert.ReplicaSetName)
```

Comment:
Let's move this up towards the top such that this is done immediately after getting a handle to resource logs.